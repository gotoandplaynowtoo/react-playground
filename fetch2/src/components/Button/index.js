import styled from 'styled-components';

const Button = styled.button`
  border: 1px solid #aaa;
  padding: 4px;
  background: #eee;
`;

export default Button;