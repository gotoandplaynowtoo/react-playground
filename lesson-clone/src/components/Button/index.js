import styled from 'styled-components';

const Button = styled.button`
  background: #eee;
  border: 1px solid #aaa;
  display: inline-block;
  padding: 4px 6px;
`;

export default Button;
