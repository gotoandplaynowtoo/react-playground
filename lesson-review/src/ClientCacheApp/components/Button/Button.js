import styled from 'styled-components';
import PropTypes from 'prop-types';

const Button = styled.button`
  background: #aaa;
  border: 1px solid #eee;
  padding: 4px 6px;
`;

Button.propTypes = {
  children: PropTypes.node
};

Button.defaultProps = {
  children: 'Button'
};

export default Button;