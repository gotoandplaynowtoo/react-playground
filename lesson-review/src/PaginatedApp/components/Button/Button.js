import styled from 'styled-components';
import PropTypes from 'prop-types';

const Button = styled.button`
  background: #eee;
  border: 1px solid #aaa;
  padding: 4px 6px;
`;

Button.propTypes = {
  children: PropTypes.node
};

export default Button;