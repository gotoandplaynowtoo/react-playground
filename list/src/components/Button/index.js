import styled from 'styled-components';

const Button = styled.button`
  padding: 4px;
  border: 1px solid #aaa;
`;

export default Button;