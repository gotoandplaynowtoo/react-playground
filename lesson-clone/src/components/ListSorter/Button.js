import Button from 'components/Button';

const XButton = Button.extend`
  border: 1px solid ${ props => props.focused ? '#222' : '#aaa'};
`;

export default XButton;
