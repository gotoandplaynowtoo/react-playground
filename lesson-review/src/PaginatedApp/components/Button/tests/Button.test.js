import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Button from 'PaginatedApp/components/Button';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches to snapshot', () => {
  const tree = renderer.create(<Button/>).toJSON();
  expect(tree).toMatchSnapshot();
});