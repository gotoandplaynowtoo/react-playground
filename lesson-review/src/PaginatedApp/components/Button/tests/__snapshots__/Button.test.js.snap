// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches to snapshot 1`] = `
<button
  className="sc-bdVaJa gnbHFc"
/>
`;
